internal/bootloader/grubenv/grubenv.go

```diff
@@ -0,0 +1,117 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2014-2015 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
```

Comment:
A bit of a disclaimer, but regarding all the `grubenv` code, it's been copied straight from [snapd](https://github.com/snapcore/snapd/blob/master/bootloader/grubenv/grubenv.go), so I'm a bit reluctant to touch it just yet since it's mostly going to end up in `x-go`. I agree some parts of it can be streamlined but I wouldn't go down that rabbithole just yet, what do you think?

Reply:
Ah, fair enough.

Reply:
You might want to mention that in the PR description.